ansible_3par_docker_plugin/tasks/install_prerequisites_on_all.yml

```diff
@@ -4,6 +4,80 @@
+    
+  - name:  Initialize valid driver list
+    set_fact: 
+      default_drivers:
+      - hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver
+      - hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver
+      - hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver
+  
+  - name: Fail when backend drivers are incorrect or mispelled 
+    fail:
+      msg: "Driver {{item}} is incorrect"
+    with_items:
+      - "{{my_driver_list}}"
+    when: >-
+        "{{item}}" not in default_drivers       
+     
+  - name: Setting set_mixed flag to true if ISCSI/FC and FileDriver are present
+    set_fact:
+      set_mixed: 'true' 
+    when: >-
+      ('hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver'  in  my_driver_list and 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver' in my_driver_list) or 
+      ('hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver'  in  my_driver_list and 'hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver' in  my_driver_list )    
+     
+  - name: Fail when DEFAULT backend driver is block and DEFAULT_FILE backend is not present in mixed configuration
+    fail :
+      msg: "DEFAULT_FILE backend is not present in mixed configuration"
+    with_items:
+      - "{{ INVENTORY.keys()}}"  
+    when:  >-
+      set_mixed ==true and 
+      (INVENTORY['DEFAULT']['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver' or INVENTORY['DEFAULT']['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver') and 
+      INVENTORY['DEFAULT_FILE'] is not defined
+      
+  - name: Fail when DEFAULT backend driver is FileDriver and DEFAULT_BLOCK backend is not present in mixed configuration
+    fail :
+      msg: "DEFAULT_BLOCK backend is not present in mixed configuration"
+    with_items:
+      - "{{ INVENTORY.keys()}}"  
+    when:  >-
+      set_mixed ==true and INVENTORY['DEFAULT']['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver'
+      and INVENTORY['DEFAULT_BLOCK'] is not defined  
+      
+  
+  - name: Fail when file driver is not present in DEFAULT_FILE backend 
+    fail :
+      msg: "DEFAULT_FILE backend driver is incorrect in mixed configuration"
+    with_items:
+      - "{{ INVENTORY.keys()}}"  
+    when:  >-
+      set_mixed ==true and INVENTORY['DEFAULT_FILE'] is defined and INVENTORY['DEFAULT_FILE']['hpedockerplugin_driver'] != 'hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver' 
+
+  - name: Fail when block driver for FC/ISCSI is not present in DEFAULT_BLOCK backend
+    fail :
+      msg: "DEFAULT_BLOCK backend driver is incorrect in mixed configuration"
+    with_items:
+      - "{{ INVENTORY.keys()}}"  
+    when:  >-
+      set_mixed ==true and INVENTORY['DEFAULT_BLOCK'] is defined and
+      (INVENTORY['DEFAULT_BLOCK']['hpedockerplugin_driver'] != 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver' and INVENTORY['DEFAULT_BLOCK']['hpedockerplugin_driver'] != 'hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver')  
```

Comment:
Shouldn't we check for defined for INVENTORY['DEFAULT_BLOCK'] here?

Reply:
Already done